### PR TITLE
Disable form inputs while testing and saving source and destination

### LINKED ISSUE #12575: When a connector is being checked, the form should not be editable
Page: `/workspaces/${workspace_id}/destination/${destination_id}/settings`

When a connector is being CHECKED, and the progress bar is displayed in the UI, the form-fields should no longer be editable.  As it stands now, if a change for the form is made mid-CHECK, it is not clear which values have been validated and persisted. 

Screen Shot 2022-05-04 at 9 50 32 AM

### PR DESCRIPTION
**Pull request by edmundito**
## What
This disables all inputs while the user is saving or testing a source or destination.

Fixes airbytehq/airbyte#12575

## How
It adds disabled fields to all the components. Additionally:
* Replaces delete button in EditorRow with real button
* Fixed input and tagged input disabled state so it doesn't hover on disable
* TypeScript cleanup: Moving types to interfaces and fix a few indirect exports

## Recommended reading order
Top to bottom

## Tests

Tested:
1. Creating a new source and destination
2. Editing a source and destination
3. Sources or destinations with conditional fields, array fields, tagged inputs
4. Run and cancel test source/destination

### CODE REVIEW

**Review by teallarson — APPROVED** [review]
That ended up needing to be passed a lot of places!

Tested locally.  The form is disabled for both Source/Destination in both create/edit mode when a test or save is in progress.

File: airbyte-webapp/src/core/form/types.ts [hunk 1]
```diff
@@ -48,3 +48,6 @@ export type { FormBlock, FormConditionItem, FormGroupItem, FormObjectArrayItem }
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 export type WidgetConfig = { [key: string]: any };
 export type WidgetConfigMap = { [key: string]: WidgetConfig };
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type FormComponentOverrideProps = Record<string, any>;
```
Comment (timroes) [inline airbyte-webapp/src/core/form/types.ts:53]: There isn't any real need making this an `any` record and thus disabling type checking on this pretty much in places it's used. We should type this properly with all overwrites we know should be applicable (i.e. `{ disabled?: boolean }`), and can extend it if we ever need more.

File: airbyte-webapp/src/views/Connector/ServiceForm/ServiceForm.tsx [hunk 2]
```diff
@@ -174,10 +174,12 @@ const ServiceForm: React.FC<ServiceFormProps> = (props) => {
   const uiOverrides = useMemo(
     () => ({
       name: {
-        component: (property: FormBaseItem) => <ConnectorNameControl property={property} formType={formType} />,
+        component: (property: FormBaseItem, componentProps: FormComponentOverrideProps) => (
```
Comment (timroes) [inline airbyte-webapp/src/views/Connector/ServiceForm/ServiceForm.tsx:177]: I'd ideally prefer if we wouldn't need that `componentProps` hack here, and instead have the `disabled` property calculated in this component, but given that we need the `isSubmitting` from the formik state, which only exists a level lower, this is unfortunately not easily achievable atm :-/ But something we might want to keep in mind for later refactorings.

File: airbyte-webapp/src/core/form/types.ts (hunk 1 above)
Comment (edmundito) [reply airbyte-webapp/src/core/form/types.ts:53]: Unfortunately, the typing in this area of the code is in really bad shape. Based on this comment I documented the current issue in #13331

**Pull request merged**